### PR TITLE
Decompiled and Matched func_802A10A4_77F2E4

### PR DESCRIPTION
**Pull request by Nasina7**
I've done a good amount of emudev work before, but this is my first time decompiling anything.  If I did anything wrong, or if there is anything I need to change, please let me know!

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.01%) on `us`.

File: src/battle/move/shell_crack/77F240.c [hunk 1]
```diff
@@ -6,4 +6,24 @@
...
+    do {
+        temp_f20 = (temp_s0->goalPos.x + (f32) rand_int(0x1E)) - 15.0f;
+        temp_s1 = ++phi_s1;
+        fx_floating_cloud_puff(0, temp_f20, (temp_s0->goalPos.y + (f32) rand_int(0x14)) - 15.0f, temp_s0->goalPos.z + 5.0f, 1.0f, 0x19);
+        phi_s1 = temp_s1;
+    } while (temp_s1 < 0x14);
+    return 2;
+}
+/* Warning: struct EffectInstance is not defined (only forward-declared) */
+/* Warning: struct ActorBlueprint is not defined (only forward-declared) */
```
Comment (nanaian) [inline src/battle/move/shell_crack/77F240.c:28]: Can remove these comments
```suggestion
```

**Review by nanaian — CHANGES_REQUESTED** [review]
Awesome first decomp work! I've made some comments, feel free to contact us in the Discord if have any trouble

File: src/battle/move/shell_crack/77F240.c [hunk 2]
```diff
@@ -6,4 +6,24 @@
 
 #include "world/common/IsRightOnEquipped.inc.c"
 
-INCLUDE_ASM(s32, "battle/move/shell_crack/77F240", func_802A10A4_77F2E4);
+extern void fx_floating_cloud_puff(int, f32, f32, f32, f32, s32); /* extern */
+
+int func_802A10A4_77F2E4(void) {
```
Comment (nanaian) [inline src/battle/move/shell_crack/77F240.c:11]: This is almost certainly a script API function (judging from the `return 2`), so please make its signature the following
```suggestion
ApiStatus func_802A10A4_77F2E4(Evt* script, s32 isInitialCall) {
```

File: src/battle/move/shell_crack/77F240.c [hunk 3]
```diff
@@ -6,4 +6,24 @@
...
+
+    temp_s0 = &gBattleStatus.playerActor->state;
+    phi_s1 = 0;
+    do {
+        temp_f20 = (temp_s0->goalPos.x + (f32) rand_int(0x1E)) - 15.0f;
+        temp_s1 = ++phi_s1;
+        fx_floating_cloud_puff(0, temp_f20, (temp_s0->goalPos.y + (f32) rand_int(0x14)) - 15.0f, temp_s0->goalPos.z + 5.0f, 1.0f, 0x19);
+        phi_s1 = temp_s1;
+    } while (temp_s1 < 0x14);
+    return 2;
```
Comment (nanaian) [inline src/battle/move/shell_crack/77F240.c:25]:
```suggestion
    return ApiStatus_DONE2;
```

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/battle/move/shell_crack/77F240.c [hunk 4]
```diff
@@ -1,29 +1,24 @@
...
-    do {
-        temp_f20 = (temp_s0->goalPos.x + (f32) rand_int(0x1E)) - 15.0f;
-        temp_s1 = ++phi_s1;
-        fx_floating_cloud_puff(0, temp_f20, (temp_s0->goalPos.y + (f32) rand_int(0x14)) - 15.0f, temp_s0->goalPos.z + 5.0f, 1.0f, 0x19);
-        phi_s1 = temp_s1;
-    } while (temp_s1 < 0x14);
-    return 2;
+    playerActorState = &gBattleStatus.playerActor->state;
+    for(i = 0; i < 20; i++)
+    {
```
Comment (nanaian) [inline src/battle/move/shell_crack/77F240.c:17]:
```suggestion
    for(i = 0; i < 20; i++) {
```

File: src/battle/move/shell_crack/77F240.c [hunk 5]
```diff
@@ -1,29 +1,24 @@
...
+        fx_floating_cloud_puff(0, 
+                               (playerActorState->goalPos.x + (f32) rand_int(30)) - 15.0f, 
+                               (playerActorState->goalPos.y + (f32) rand_int(20)) - 15.0f, 
+                               playerActorState->goalPos.z + 5.0f, 1.0f, 25);   
+    }
```
Comment (nanaian) [inline src/battle/move/shell_crack/77F240.c:22]:
```suggestion
    }

```

**Review by nanaian — CHANGES_REQUESTED** [review]
Style nits; LGTM otherwise

**Comment by Nasina7** [comment]
Accidentally reverted those changes while fixing the fx_floating_cloud_puff issue.  Should be fixed now.

File: src/battle/move/shell_crack/77F240.c [hunk 6]
```diff
@@ -1,9 +1,24 @@
...
 
 #include "world/common/IsRightOnEquipped.inc.c"
 
-INCLUDE_ASM(s32, "battle/move/shell_crack/77F240", func_802A10A4_77F2E4);
+ApiStatus func_802A10A4_77F2E4(Evt* script, s32 isInitialCall) {
+    ActorState* playerActorState;
+    s32 i;
+
+    playerActorState = &gBattleStatus.playerActor->state;
+    for(i = 0; i < 20; i++) {
```
Comment (nanaian) [inline src/battle/move/shell_crack/77F240.c:16]:
```suggestion
    for (i = 0; i < 20; i++) {
```

**Review by nanaian — CHANGES_REQUESTED** [review]
oop

**Comment by Nasina7** [comment]
:+1:

File: src/battle/move/shell_crack/77F240.c [hunk 7]
```diff
@@ -1,9 +1,24 @@
...
+
 #define NAMESPACE battle_move_shell_crack
 
 #include "world/common/IsBerserkerEquipped.inc.c"
 
 #include "world/common/IsRightOnEquipped.inc.c"
 
-INCLUDE_ASM(s32, "battle/move/shell_crack/77F240", func_802A10A4_77F2E4);
+ApiStatus func_802A10A4_77F2E4(Evt* script, s32 isInitialCall) {
+    ActorState* playerActorState;
```
Comment (ethteck) [inline src/battle/move/shell_crack/77F240.c:12]:
```suggestion
    ActorState* playerActorState = &gBattleStatus.playerActor->state;
```

File: src/battle/move/shell_crack/77F240.c [hunk 8]
```diff
@@ -1,9 +1,24 @@
...
+    s32 i;
+
+    playerActorState = &gBattleStatus.playerActor->state;
```
Comment (ethteck) [inline src/battle/move/shell_crack/77F240.c:15]: this line can be moved into the declaration (as shown above)

**Review by ethteck — CHANGES_REQUESTED** [review]
Looks great! Just this small change and should be good to merge. and welcome :)

**Pull request merged**